### PR TITLE
[Quantum RPG] Add help and descriptions to qaracter actions

### PR DESCRIPTION
**Pull request by dstrain115**
Now, the qaracter actions are explained, such as:
`s) Sample enemy qubit.`

There is also a help option within the battle.

### CODE REVIEW

File: unitary/examples/quantum_rpg/battle_test.py [hunk 1]
```diff
@@ -107,9 +104,32 @@ def test_battle_loop():
...
+-----------------------------------------------
+Aaronson Analyst   watcher Observer
+1QP (0|1> 0|0> 1?)   1QP (0|1> 0|0> 1?)
+-----------------------------------------------
+Aaronson turn:
+m) Measure enemy qubit.
+h) Help.
+The analyst can measure enemy qubits.  This forces an enemy qubit
+into the |0> state or |1> state with a probability based on its
+amplitude. Try to measure the enemy qubits as |0> to defwat them.
```
Comment (cognigami) [inline unitary/examples/quantum_rpg/battle_test.py:133]: defwat
defeat

File: unitary/examples/quantum_rpg/classes.py [hunk 2]
```diff
@@ -24,7 +24,18 @@ class Engineer(qaracter.Qaracter):
     Engineeers can do an X gate at level 1 and a H gate at level 3.
     """
 
+    def action_descriptions(self) -> Dict[str, str]:
+        """Descriptions for each player action."""
+        return {"x": "Attack with X gate"}
+
+    def help(self) -> str:
+        return (
+            "The Engineer can 'attack' by applying an X gate to an enemw'y qubit.\n"
```
Comment (cognigami) [inline unitary/examples/quantum_rpg/classes.py:33]: enemw'y
enemy

File: unitary/examples/quantum_rpg/classes.py [hunk 3]
```diff
@@ -41,10 +52,32 @@ class Analyst(qaracter.Qaracter):
...
     """
 
+    def action_descriptions(self) -> Dict[str, str]:
+        """Descriptions for each player action."""
+        return {"s": "Sample enemy qubit", "m": "Measure enemy qubit"}
+
+    def help(self) -> str:
+        msg = "The analyst can measure enemy qubits.  This forces an enemy qubit\n"
+        msg += "into the |0> state or |1> state with a probability based on its\n"
+        msg += "amplitude. Try to measure the enemy qubits as |0> to defwat them.\n"
```
Comment (cognigami) [inline unitary/examples/quantum_rpg/classes.py:62]: defwat
defeat

**Review by cognigami — CHANGES_REQUESTED** [review]

**Review by cognigami — APPROVED** [review]

File: unitary/examples/quantum_rpg/classes.py (hunk 3 above)
Comment (dstrain115) [reply unitary/examples/quantum_rpg/classes.py]: Done.

File: unitary/examples/quantum_rpg/classes.py (hunk 2 above)
Comment (dstrain115) [reply unitary/examples/quantum_rpg/classes.py]: Done.

File: unitary/examples/quantum_rpg/battle_test.py (hunk 1 above)
Comment (dstrain115) [reply unitary/examples/quantum_rpg/battle_test.py]: Done.

**Pull request merged**